src/ordered-elements.js

```diff
@@ -1,35 +1,61 @@
 /* @flow */
-/* global Map */
+const MAP_EXISTS = typeof Map !== 'undefined';
 
 export default class OrderedElements {
     /* ::
     elements: {[string]: any};
     keyOrder: string[];
-
-    static fromObject: ({[string]: any}) => OrderedElements;
-    static fromMap: (Map<string,any>) => OrderedElements;
-    static from: (Map<string,any> | {[string]: any} | OrderedElements) =>
-        OrderedElements;
     */
 
-    constructor(
-        elements /* : {[string]: any} */ = {},
-        keyOrder /* : string[] */ = []
-    ) {
-        this.elements = elements;
-        this.keyOrder = keyOrder;
+    constructor() {
+        this.elements = {};
+        this.keyOrder = [];
     }
 
     forEach(callback /* : (string, any) => void */) {
         for (let i = 0; i < this.keyOrder.length; i++) {
-            callback(this.keyOrder[i], this.elements[this.keyOrder[i]]);
+            // (value, key) to match Map's API
+            callback(this.elements[this.keyOrder[i]], this.keyOrder[i]);
         }
     }
 
     set(key /* : string */, value /* : any */) {
         if (!this.elements.hasOwnProperty(key)) {
             this.keyOrder.push(key);
         }
+
```

Comment:
It seems non-intuitive that `.set()` does a deep merge of the elements. Could we maybe pull this deep merge into a separate method, or if this isn't being used anywhere else, rename this?